tests/test_gbs.py

```diff
@@ -374,25 +373,25 @@ class TestStrawberryFieldsGBS:
     """Unit tests for StrawberryFieldsGBS."""
 
     @pytest.mark.parametrize("use_cache", [True, False])
-    @pytest.mark.parametrize("analytic", [True, False])
+    @pytest.mark.parametrize("shots", [None, 1000])
     @pytest.mark.parametrize("cutoff", [3, 6])
-    def test_load_device(self, cutoff, analytic, use_cache):
+    def test_load_device(self, cutoff, shots, use_cache):
         """Test that the device loads correctly"""
         dev = qml.device(
             "strawberryfields.gbs",
             wires=2,
             cutoff_dim=cutoff,
-            analytic=analytic,
+            shots=shots,
             use_cache=use_cache,
         )
         assert dev.cutoff == cutoff
-        assert dev.analytic is analytic
+        assert dev.analytic is (shots is None)
         assert dev.use_cache is use_cache
 
     def test_load_device_with_samples(self):
         """Test that the device loads correctly with input samples"""
         a = np.ones((10, 2))
-        dev = qml.device("strawberryfields.gbs", wires=2, cutoff_dim=3, analytic=False, samples=a)
+        dev = qml.device("strawberryfields.gbs", wires=2, cutoff_dim=3, shots=10, samples=a)
```

Comment:
This might highlight a subtlety in the device that `shots` doesn't need to be set to 10. It could be set to anything as long as `shots is not None` (i.e., we are in "non-analytic mode").

Overall, I think this device did lean on the `analytic` keyword argument a bit more than usual. It is probably ok though - not sure if this device is used much and things still work if you set shots to be the number of samples (as you have done above)

Reply:
Yeah, I noticed this while updating the tests. Did you want to add validation to ensure that `shots==len(samples)`, or do you want to leave it as is?

Alternatively, you could allow `shots<=len(samples)`, and have it truncate the samples if shorter.